webapp/channels/src/components/create_team/components/team_url/team_url.test.tsx

```diff
@@ -65,15 +50,28 @@ describe('/components/create_team/components/display_name', () => {
         const actions = {...defaultProps.actions, checkIfTeamExists};
         const props = {...defaultProps, actions};
 
-        const wrapper = mountWithIntl(
+        renderWithContext(
             <TeamUrl {...props}/>,
         );
 
-        await (wrapper.instance() as unknown as TeamUrl).submitNext({preventDefault: jest.fn()} as unknown as React.MouseEvent<Button, MouseEvent>);
+        screen.getByText('Finish').click();
+
+        // Wait for the check to see if the team already exists which should fail
+        await Promise.resolve();
+
+        expect(screen.getByText('This URL is taken or unavailable. Please try another.')).toBeInTheDocument();
```

Comment:
Instead of doing this `await Promise.resolve` can we use some selector that waits for the element to be in screen? I am not sure if `getByText` does that (maybe it does) but I think there was some selector that waited for the element you are looking for to appear on screen.

Reply:
There is `findByText` that does the waiting, so I'll change it to that

Reply:
Actually, I was reminded by the the internet that [https://testing-library.com/docs/dom-testing-library/api-async/#waitfor](`waitFor`) exists and is perfect for both this case and the ones above